sse2neon.h

```diff
@@ -8358,6 +8359,638 @@ FORCE_INLINE int _mm_testz_si128(__m128i a, __m128i b)
+            vceqq_u8(vdupq_n_u8(vgetq_lane_u8(vreinterpretq_u8_m128i(b), 5)),  \
+                     vreinterpretq_u8_m128i(a)));                              \
+        mtx[6] = vreinterpretq_m128i_u8(                                       \
+            vceqq_u8(vdupq_n_u8(vgetq_lane_u8(vreinterpretq_u8_m128i(b), 6)),  \
+                     vreinterpretq_u8_m128i(a)));                              \
+        mtx[7] = vreinterpretq_m128i_u8(                                       \
+            vceqq_u8(vdupq_n_u8(vgetq_lane_u8(vreinterpretq_u8_m128i(b), 7)),  \
+                     vreinterpretq_u8_m128i(a)));                              \
+        mtx[8] = vreinterpretq_m128i_u8(                                       \
+            vceqq_u8(vdupq_n_u8(vgetq_lane_u8(vreinterpretq_u8_m128i(b), 8)),  \
+                     vreinterpretq_u8_m128i(a)));                              \
+        mtx[9] = vreinterpretq_m128i_u8(                                       \
+            vceqq_u8(vdupq_n_u8(vgetq_lane_u8(vreinterpretq_u8_m128i(b), 9)),  \
+                     vreinterpretq_u8_m128i(a)));                              \
+        mtx[10] = vreinterpretq_m128i_u8(                                      \
+            vceqq_u8(vdupq_n_u8(vgetq_lane_u8(vreinterpretq_u8_m128i(b), 10)), \
+                     vreinterpretq_u8_m128i(a)));                              \
+        mtx[11] = vreinterpretq_m128i_u8(                                      \
+            vceqq_u8(vdupq_n_u8(vgetq_lane_u8(vreinterpretq_u8_m128i(b), 11)), \
+                     vreinterpretq_u8_m128i(a)));                              \
+        mtx[12] = vreinterpretq_m128i_u8(                                      \
+            vceqq_u8(vdupq_n_u8(vgetq_lane_u8(vreinterpretq_u8_m128i(b), 12)), \
+                     vreinterpretq_u8_m128i(a)));                              \
+        mtx[13] = vreinterpretq_m128i_u8(                                      \
+            vceqq_u8(vdupq_n_u8(vgetq_lane_u8(vreinterpretq_u8_m128i(b), 13)), \
+                     vreinterpretq_u8_m128i(a)));                              \
+        mtx[14] = vreinterpretq_m128i_u8(                                      \
+            vceqq_u8(vdupq_n_u8(vgetq_lane_u8(vreinterpretq_u8_m128i(b), 14)), \
+                     vreinterpretq_u8_m128i(a)));                              \
+        mtx[15] = vreinterpretq_m128i_u8(                                      \
+            vceqq_u8(vdupq_n_u8(vgetq_lane_u8(vreinterpretq_u8_m128i(b), 15)), \
+                     vreinterpretq_u8_m128i(a)));                              \
+    } while (0)
+
+#define PCMPSTR_RNG_U16x8(a, b, mtx)                                           \
+    do {                                                                       \
+        uint16x8_t vec_b[8];                                                   \
+        __m128i mask = vreinterpretq_m128i_u32(vdupq_n_u32(0xffff));           \
```

Comment:
Instead of manually unrolling loops, we can explicitly use GNU extension `#pragma unroll`.
See [Arm Compiler for Embedded User Guide: Optimizing loops](https://developer.arm.com/documentation/100748/0618/Writing-Optimized-Code/Optimizing-loops)

Reply:
So after the compilation on ARMv8-A and ARMv7-A, it seems that `vgetq_lane_xxx` needs constant intermediate. Taking ARMv7-A as example:
```
In function ‘uint8_t vgetq_lane_u8(uint8x16_t, int)’,
    inlined from ‘int _sse2neon_cmp_byte_equal_any(__m128i, int, __m128i, int)’ at ./sse2neon.h:8558:5:
/usr/lib/gcc-cross/arm-linux-gnueabihf/9/include/arm_neon.h:6274:50: error: argument 2 must be a constant immediate
 6274 |   return (uint8_t)__builtin_neon_vget_laneuv16qi ((int8x16_t) __a, __b);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
      ```